plugins/module_utils/dnac.py

```diff
@@ -992,6 +993,182 @@ def check_tasks_response_status(self, response, api_name):
+                return None
+
+        except Exception as e:
+            # Log an error message and fail if an exception occurs
+            self.log_traceback()
+            self.msg = (
+                "An error occurred while executing API call {0} - {1}. "
+                "Parameters: {2}. Error: {3}.".format(api_family, api_function, api_parameters, str(e))
+            )
+            self.fail_and_exit(self.msg)
+
+    def get_task_status_from_taskid(self, **kwargs):
+        """
+        Retrieves and monitors the status of a task by its task ID.
+
+        This function continuously checks the status of a specified task using its task ID.
+        If the task completes successfully, it updates the message and status accordingly.
+        If the task fails or times out, it handles the error and updates the status and message.
+
+        Parameters:
+        - task_id (str): The unique identifier of the task to monitor.
+        - task_name (str): The name of the task being monitored.
+        - params (dict): Additional parameters related to the task.
+        - msg (str): The success message to set if the task completes successfully.
+
+        Returns:
+        - self: The instance of the class with updated status and message.
+        """
+        task_id = kwargs.get('task_id')
+        task_name = kwargs.get('task_name')
+        params = kwargs.get('params')
+        msg = kwargs.get('msg')
+
+        start_time = time.time()
+
+        while True:
+            response = self.get_tasks_by_id(task_id)
+
+            # Check if response is returned
+            if not response:
+                self.msg = "Error retrieving task status for {0} with task_id {1}".format(task_name, task_id)
+                self.update_result("failed", False, self.msg, "ERROR")
+                break
+
+            # Check if the elapsed time exceeds the timeout
+            if self.exit_while_loop(start_time, task_id, task_name, response):
+                break
+
+            # Handle error if task execution encounters an error
+            if response.get("status") == "FAILURE" and response.get("endTime"):
+                get_task_details_response = self.get_task_details_by_id(task_id)
+                failure_reason = get_task_details_response.get("failureReason", "Unknown reason")
+                self.msg = (
+                    "An error occurred while performing {0} task for params: {1}. "
+                    "The operation failed due to the following reason: {2}".format(
+                        task_name, params, failure_reason
+                    )
+                )
+                self.update_result("failed", False, self.msg, "ERROR")
+                break
+
+            # Check if task completed successfully based on progress_string
+            if response.get("status") == "SUCCESS" and response.get("endTime"):
```

Comment:
Can we simplify the logic? 
In current code we are getting task_status_response.get("status") two times..  Also task_status_response.get("endTime"): checked for twice and it is common.. 
```

        # Check if the task has completed (either success or failure)
        if end_time:
            if status == "FAILURE":

                break
            elif status == "SUCCESS":

                break
```